ci/src/main/java/ci/ContinuousIntegrationServer.java

```diff
@@ -176,57 +119,95 @@ public void generateBuildLog(PushStatus ps) {
             date = date.replace(' ', 'T');
             String buildLogName = date + ".txt";
             FileWriter logWriter = new FileWriter("../buildHistory/" + buildLogName);
-    
+
             // write the log information to file
             logWriter.write("Build date: " + ps.getCommitDate() + "\n");
             logWriter.write(Email.getContent(ps));
-            
+
             logWriter.close();
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             e.printStackTrace();
         }
     }
 
     /**
-     * Parse request data and create a string containing the payload.
-     *
-     * @param request A HttpServletRequest recieved by handler.
-     * @return String Payload of the request.
+     * @param url         Requested URL
+     * @param baseRequest A request instance is created for each HttpConnection accepted by the server
+     * @param response    ServletResponse
+     * @return Boolean
+     * @throws IOException general class of exceptions
      */
-    public String getRequestPayload(HttpServletRequest request) {
-        StringBuilder stringBuilder = new StringBuilder();
-        BufferedReader bufferedReader = null;
+    public boolean buildInUrl(String url, Request baseRequest, HttpServletResponse response) throws IOException {
```

Comment:
I think you can divide the function more, e.g. changing lines 144-153 into a function "isSingleBuild"